src/devices/Card/CreditCard/ProcessError.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.Card.CreditCardProcessing
+{
+    /// <summary>
+    /// Processing error class
+    /// </summary>
+    public class ProcessError
+    {
+        /// <summary>
+        /// The Error type
+        /// </summary>
+        public ErrorType ErrorType { get; internal set; }
+
+        /// <summary>
+        /// Complementary data for some errors
+        /// </summary>
+        public byte CorrectLegnthOrBytesAvailable { get; internal set; }
+
+        /// <summary>
+        /// Constructor to process the error
+        /// </summary>
+        /// <param name="errorToProcess">A span of byte</param>
+        public ProcessError(Span<byte> errorToProcess)
+        {
+            // EMV 4.3 Book 3 page 44
+            if (errorToProcess.Length < 2)
+                ErrorType = ErrorType.Unknown;
+            else if ((errorToProcess[0] == 0x90) && (errorToProcess[0] == 0x90))
```

Comment:
is it possible to construct short from the first two elements and then convert most of those to switch statement or just cast them directly to enum (except for those few values in the bottom where you check only first byte)

Reply:
Yea, that should be douable in some ways. See EMV_v4.3_Book_3page 60 and the following ones. It's jsut a summary but that will give you an idea.